core/string/ustring.cpp

```diff
@@ -4324,9 +4326,8 @@ String String::path_to(const String &p_path) const {
 
 	String dir;
 
-	for (int i = src_dirs.size() - 1; i > common_parent; i--) {
-		dir += "../";
-	}
+	int dirs_to_backtrack = (src_dirs.size() - 1) - common_parent;
+	dir = String("../").repeat(dirs_to_backtrack);
```

Comment:
If changing `+=` to `=` then I'd also move the declaration in here (remove the one above).
```suggestion
	String dir = String("../").repeat(dirs_to_backtrack);
```